test/configparser.cpp

```diff
@@ -106,4 +106,17 @@ TEST_CASE("include directive includes other config files", "[ConfigParser]")
 	SECTION("Diamond of death includes pass") {
 		REQUIRE_NOTHROW(cfgparser.parse("data/diamond-of-death/A"));
 	}
+	SECTION("Recursive include don't happen") {
+		TestHelpers::TempFile testfile;
+		TestHelpers::EnvVar("$TMPFILE").set(testfile.getPath()); // $TMPFILE used in conf file
+
+		REQUIRE_NOTHROW(cfgparser.parse("data/recursive-include-side-effect")); // recursive includes don't fail
+		// I think it will never get below here and fail? If it recurses, the above fails
+
+		std::ifstream in(testfile.getPath(), std::ifstream::ate | std::ifstream::binary);
+		REQUIRE(in.tellg() < 100); // 100 is just a reasonably high number for one command output
+
+		// TODO: timeout somehow? wouldn't want the tests going on forever if it
+		// actually is recursing
+	}
```

Comment:
The expected behaviour is that the include happens only once, right? It's better to spell that out in the test description, since "don't happen" doesn't exactly tell the reader what *does* happen.

Same goes for the `REQUIRE`: if we expect exactly one line in the file, it's better to check *that*, rather than assuming that a single call to `date` won't print out more than 100 bytes, or that 2+ calls will definitely print out more than 100. (I find that with tests, the dumber code the better.)

I don't have a solution for the timeout thing. What I do, personally, is run tests with `/bin/timeout 3 tests/tests`. It's a bit of a kludge, but it works and I can only recall one bug that did hang the tests.

Reply:
Alright. commits fd1fd74 and fd02d4c should deal with these. Turns out the test wasn't actually working, but now it should be